config.go

```diff
@@ -84,6 +86,51 @@ func setupAllowedPatterns() {
 	}
 }
 
+
+func setupRemoteAuth() {
+	logger := log.WithField("remote_auth", *remoteAuthStr)
+
+	// Are we using any SMTP auth?
```

Comment:
Can we convert the switch to an simple if - or move/change/remove the comment? The comment indicates what we are doing is checking if auth is used - but the switch indicates that we're doing some early checks for individual auth methods.

Reply:
Er, yeah I see how this was a little confusing. I liked using the `switch` because I could check two cases at once (`""` and `"none"). Would you prefer:
1. Changing the comment:
    ```go
        // Is remote auth disabled?
        switch *remoteAuthStr {
        case "", "none":
    ```
1. Changing the comment and using an `if`:
    ```go
        // Is remote auth disabled?
        if *remoteAuthStr == "" || *remoteAuthStr == "none" {
    ```
1. Dropping support for `""` and requiring `"none"`
    ```go
        // Is remote auth disabled?
        if *remoteAuthStr == "none" {
    ```

It's a little tricky to write this function w/o duplicating all of the cases, or duplicating the username/password check.

I updated the code with option 1. but I'd be happy to use whatever style you think is best.

Reply:
Hey, just a friendly reminder that this discussion is outstanding.

Reply:
I would prefer variant 2 because comment and code are consistent there but I have no strong opinion there.

Reply:
Okay, no problem. I'll fix this up before we merge.

Reply:
Changed in 3debf4127d364764789eae1bee9564080219a789.